### PR TITLE
[IMP] spreadsheet: allow to reorder global filters

### PR DESCRIPTION
**Pull request by hokolomopo**
## [IMP] spreadsheet: allow to reorder global filters

This commit allows to reorder global filters via drag and drop. The
fact that global filters have a concept of order means taht the
structure of the global filters in the core plugin need to change from
an Object<string, GlobalFilter> to an Array<GlobalFilter>.

## [REF] spreadsheet_edition: de-duplicate filter id in cmds

The commands `ADD_GLOBAL_FILTER` and `EDIT_GLOBAL_FILTER` had the filter id
once in cmd.id, and once in cmd.filter.id. This made it possible to send
a command with 2 different ids, which can have arbitrary behaviour depending
on the implementation details of the command.

Changed it so we only use cmd.filter.id, and removed the cmd.id field. Also
simplified the commands helpers for the tests.

Task: [3502684](https://www.odoo.com/web#id=3502684&menu_id=4720&cids=1&action=333&active_id=2328&model=project.task&view_type=form)

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/spreadsheet/static/src/global_filters/plugins/global_filters_core_plugin.js [hunk 1]
```diff
@@ -32,8 +32,8 @@ import { escapeRegExp } from "@web/core/utils/strings";
 export class GlobalFiltersCorePlugin extends spreadsheet.CorePlugin {
     constructor(config) {
         super(config);
-        /** @type {Object.<string, GlobalFilter>} */
-        this.globalFilters = {};
+        /** @type {Array.<GlobalFilter>} */
+        this.globalFilters = [];
```
Comment (anhe-odoo) [inline addons/spreadsheet/static/src/global_filters/plugins/global_filters_core_plugin.js:36]: Just a question, but why did you change this structure ? I suppose it's to keep a real order, but there is any other reason ?

File: addons/spreadsheet/static/src/global_filters/plugins/global_filters_ui_plugin.js [hunk 2]
```diff
@@ -4,7 +4,7 @@
  * @typedef {import("@spreadsheet/data_sources/metadata_repository").Field} Field
  * @typedef {import("./global_filters_core_plugin").GlobalFilter} GlobalFilter
  * @typedef {import("./global_filters_core_plugin").FieldMatching} FieldMatching
- 
+
```
Comment (anhe-odoo) [inline addons/spreadsheet/static/src/global_filters/plugins/global_filters_ui_plugin.js:7]: Empty line ?

**Review by anhe-odoo — COMMENTED** [review]
Some questions, else LGTM :)

File: addons/spreadsheet/static/src/global_filters/plugins/global_filters_core_plugin.js [hunk 3]
```diff
@@ -56,10 +56,21 @@ export class GlobalFiltersCorePlugin extends spreadsheet.CorePlugin {
                 }
                 break;
             case "ADD_GLOBAL_FILTER":
-                if (this._isDuplicatedLabel(cmd.id, cmd.filter.label)) {
+                if (this._isDuplicatedLabel(cmd.filter.id, cmd.filter.label)) {
                     return CommandResult.DuplicatedFilterLabel;
                 }
                 return checkFiltersTypeValueCombination(cmd.filter.type, cmd.filter.defaultValue);
+            case "MOVE_GLOBAL_FILTER": {
+                const index = this.globalFilters.findIndex((filter) => filter.id === cmd.id);
```
[the remark this answers is not in the document]
Comment (hokolomopo) [reply addons/spreadsheet/static/src/global_filters/plugins/global_filters_core_plugin.js:64]: `REMOVE_GLOBAL_FILTER` also have cmd.id.

File: addons/spreadsheet/static/src/global_filters/plugins/global_filters_core_plugin.js (hunk 1 above)
Comment (hokolomopo) [reply addons/spreadsheet/static/src/global_filters/plugins/global_filters_core_plugin.js:36]: Yeah it's for the order, we cannot have ordering while using a dictionnary

**Comment by rrahir** [comment]
@robodoo r+ rebase-ff
